core-foundation/src/filedescriptor.rs

```diff
@@ -81,6 +81,17 @@ impl AsRawFd for CFFileDescriptor {
     }
 }
 
+use runloop::{CFRunLoopSource};
+
+impl CFRunLoopSource {
+    pub fn from_file_descriptor(fd: &CFFileDescriptor, order: CFIndex) -> CFRunLoopSource {
+        unsafe {
+            let source_ref = CFFileDescriptorCreateRunLoopSource(kCFAllocatorDefault, fd.0, order);
```

Comment:
This can return null according to [the docs](https://developer.apple.com/documentation/corefoundation/1477593-cffiledescriptorcreaterunloopsou?language=objc).